### ISSUE TITLE
Make search better

### ISSUE THREAD

**Issue by meshulam** [codaco/feedback-site#8]
Currently the search feature just looks for exact string matches in titles and descriptions. Should it be fuzzier?

**Comment by berniehogan** [comment]
So fuzzy matching is better. Check for a https://en.wikipedia.org/wiki/Levenshtein_distance 

But, I also noticed that the search didn't search the text of my comments. I think that is a greater priority.

**Comment by jthrilly** [comment]
There are two forms of search, I think. The one that needed fuzzy matching was aimed primarily at finding existing issues to avoid duplication. This should be part of the new feedback flow. I will create a separate issue to discuss this.

The other is just a text search, which should include post content as Bernie says. This is the main "search" box on the main page. If possible, it might make sense to let the user decide between searching by title or by post content.

**Comment by meshulam** [comment]
Need to look into adding fuzzy matching for search, first pass estimate 4-8 hours.

I'd prefer to keep a simpler interface rather than adding options for how to search, but let me know if you still want to add the option for search by title/body. Would it just be a checkbox next to the existing search-as-you-type field, or a more drastic UI change?

**Comment by jthrilly** [comment]
I don't think we need that level of granularity in the UI. I think we just search titles.

**Comment by meshulam** [comment]
I've updated the app to use MongoDB's [text search](https://docs.mongodb.com/manual/text-search/), boosting posts where terms appear in the title, but still including posts that have terms in the body.

Instead of a general purpose fuzzy match, it uses some knowledge of the english language to treat singular/plural as synonyms, and some other logic related to abbreviations. Then it's basically an 'OR' on all the terms, sorting based on where the most terms appear.

I'd like people to test out the search and let me know what they think. I want to avoid introducing a true search service like ElasticSearch because it would be overkill, but if there are specific problems we should be able to tweak the results somehow.

**Comment by jthrilly** [comment]
That sounds great! I'll ask the team to test search.
